### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by jesdazrez**
Hi!

I noticed some clippy warnings while the AUR package was getting installed on my computer, so I fixed them.

Let me know if you want some adjustments :).

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -35,16 +35,17 @@ pub enum RawColor {
 }
 
 impl From<RawColor> for Color {
+    #[must_use]
```
Comment (dbrgn) [inline src/config.rs]: Hm, I don't see a big reason for these must use annotations. Also, the docs say:

> Not bad at all, this lint just shows places where you could add the attribute.

I'd remove all `#[must_use]` annotations and add `#[allow(clippy::must_use_candidate)]` to `src/main.rs` instead. (There are some more cases further down.)

Comment (jesdazrez) [reply src/config.rs]: Ok, makes sense for me. In fact, after updating to 1.41, clippy no longer lints those.

**Comment by jesdazrez** [comment]
Hi, sorry for the delay. I just updated the PR.

Thanks for the time you took to review it!

**Comment by dbrgn** [comment]
Ugh, for some reason GitHub CI doesn't seem to run for forked pull requests. I fixed it in master and will push a merge commit to your branch.

**Comment by dbrgn** [comment]
Ok, that's what I feared. Some of the features you used are not yet available on Rust 1.36. Unfortunately I did not notice those issues initially, because CI did not run.

I try to support the lowest stable Rust version that still works, because some distros package a Rust version that's not the latest.

Could you "backtrack" the changes until the CI passes again? I think the main issue is using `Self::` in enum matches.

(If you don't have time to do so, you can also let me know and I'll handle the necessary changes.)

**Comment by jesdazrez** [comment]
Oops, my mistake for not noticing the MSRV in the README.

Let me try with 1.36. I will update the PR as soon as I can.

**Comment by jesdazrez** [comment]
Done :)

**Comment by dbrgn** [comment]
Great, thanks!

**Pull request merged**